### PR TITLE
Increased test coverage

### LINKED ISSUE #1575: Add tests to cover cases in financial aid frontend code
Specifically:
- skip dialog link when user is auto-approved for a price
- docs visibility link
- cancel button on `FinancialAidCard`

The 'Enroll and pay later' link in the dashboard also should be covered

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #1575

#### What's this PR do?
Adds or fixes tests to increase the test coverage

#### How should this be manually tested?
In addition to adding tests this also removes some dead code from `FinancialAidCalculator`, so make sure the skip links still work.

Run `npm run coverage` and verify that the relevant cases in #1575 are covered.

### CODE REVIEW

**Comment by aliceriot** [comment]
skip link works! reading code now.

File: static/js/components/dashboard/FinancialAidCard_test.js [hunk 1]
```diff
@@ -182,4 +190,13 @@ describe("FinancialAidCard", () => {
       }
     });
   });
+
+  it('hides the skip dialog if the cancel button is clicked', () => {
+    let program = programWithStatus();
+    let setConfirmSkipDialogVisibility = sandbox.stub();
+    let wrapper = renderCard({ program, setConfirmSkipDialogVisibility });
+    let props = wrapper.find("SkipFinancialAidDialog").props();
+    props['cancel']();
```
Comment (aliceriot) [inline static/js/components/dashboard/FinancialAidCard_test.js]: why not `props.cancel()` here?

Comment (aliceriot) [inline static/js/components/dashboard/FinancialAidCard_test.js]: I might even just do

```js
wrapper.find('SkipFinancialAidDialog').props().cancel();
```

File: static/js/containers/App_test.js [hunk 2]
```diff
@@ -179,4 +182,43 @@ describe('App', () => {
...
+
+          return listenForActions([SET_NAV_DRAWER_OPEN], () => {
+            node.simulate('click');
+          }).then(state => {
+            assert.isFalse(state.ui.navDrawerOpen);
+          });
+        });
+      });
+    }
+  });
```
Comment (aliceriot) [inline static/js/containers/App_test.js]: :100: it's great to have tests for this stuff!

File: static/js/containers/FinancialAidCalculator.js [hunk 3]
```diff
@@ -278,7 +270,6 @@ const mapDispatchToProps = dispatch => {
   return {
     closeDialogAndCancel: closeDialogAndCancel(dispatch),
     saveFinancialAid: saveFinancialAid(dispatch),
-    skipFinancialAid: skipFinancialAidHelper(dispatch),
```
Comment (aliceriot) [inline static/js/containers/FinancialAidCalculator.js]: code deletion === yay!

File: static/js/lib/api_test.js [hunk 4]
```diff
@@ -80,11 +79,7 @@ describe('api', function() {
 
     it('patches a user profile', () => {
       fetchJSONStub.returns(Promise.resolve(USER_PROFILE_RESPONSE));
-      fetchMock.mock('/api/v0/profiles/jane/', (url, opts) => {
-        assert.deepEqual(JSON.parse(opts.body), USER_PROFILE_RESPONSE);
-        return { status: 200 };
-      });
-      return patchUserProfile('jane', USER_PROFILE_RESPONSE).then(returnedProfile => {
+      patchUserProfile('jane', USER_PROFILE_RESPONSE).then(returnedProfile => {
```
Comment (aliceriot) [inline static/js/lib/api_test.js]: what's the reason for this change?

Comment (aliceriot) [inline static/js/lib/api_test.js]: (and below)

Comment (aliceriot) [inline static/js/lib/api_test.js]: Ah, looks like you moved these `fetchMock` assertions into some more generalized tests down below - makes sense.

File: static/js/lib/api_test.js [hunk 5]
```diff
@@ -392,84 +361,174 @@ describe('api', function() {
...
+        document.cookie = `csrftoken=${CSRF_TOKEN}`;
+      });
+
+      it('fetches and populates appropriate headers for GET', () => {
+        let body = "body";
+
+        fetchMock.mock('/url', (url, opts) => {
+          assert.deepEqual(opts, {
+            credentials: "same-origin",
+            headers: {},
```
Comment (aliceriot) [inline static/js/lib/api_test.js]: should this have the CSRF token in the headers?

File: static/js/lib/api_test.js [hunk 6]
```diff
@@ -392,84 +361,174 @@ describe('api', function() {
...
+            status: 200,
+            body: '{"json": "here"}'
+          };
         });
 
-        return assert.isRejected(fetchJSONWithCSRF('/url'));
+        return fetchJSONWithCSRF('/url', {
+          method: 'PATCH',
+          body: JSON.stringify(expectedJSON)
+        }).then(text => {
```
Comment (aliceriot) [inline static/js/lib/api_test.js]: should `text` be `responseBody` or similar?

File: static/js/containers/App_test.js [hunk 7]
```diff
@@ -179,4 +182,43 @@ describe('App', () => {
       });
     });
   });
+
+  describe('navbar', () => {
+    for (const [title, url] of [
+      ['Dashboard', '/dashboard'],
+      ['View Profile', `/learner/${SETTINGS.user.username}`],
```
Comment (singingwolfboy) [inline static/js/containers/App_test.js]: Could it ever be the case that `SETTINGS.user` is null? That is, an anonymous user is viewing this page?

File: static/js/containers/App_test.js [hunk 8]
```diff
@@ -179,4 +182,43 @@ describe('App', () => {
...
+      ['Settings', '/settings'],
+    ]) {
+      it(`closes the drawer and changes the URL when ${title} is clicked`, () => {
+        helper.store.dispatch(setNavDrawerOpen(true));
+        return renderComponent("/").then(([wrapper]) => {
+          let node = wrapper.find(".nav-drawer").find("Link").filterWhere(x => x.text() === title);
+          assert.equal(node.props()['to'], url);
```
Comment (singingwolfboy) [inline static/js/containers/App_test.js]: Why not `node.props().to` ?

File: static/js/containers/App_test.js (hunk 7 above)
Comment (noisecapella) [inline static/js/containers/App_test.js]: Not in this test since that value is set in `global_init.js`

File: static/js/lib/api_test.js (hunk 6 above)
Comment (noisecapella) [inline static/js/lib/api_test.js]: Will do

File: static/js/lib/api_test.js (hunk 5 above)
Comment (noisecapella) [inline static/js/lib/api_test.js]: No, for GET we don't need it

File: static/js/lib/api_test.js (hunk 4 above)
Comment (noisecapella) [inline static/js/lib/api_test.js]: Yeah, they weren't even being used here since we mocked `fetchJSONWithCSRF`

**Comment by noisecapella** [comment]
Comments addressed

File: static/js/lib/api_test.js (hunk 4 above)
Comment (aliceriot) [inline static/js/lib/api_test.js]: oh lol that's goofy. glad you caught that!

File: static/js/lib/api_test.js (hunk 5 above)
Comment (aliceriot) [inline static/js/lib/api_test.js]: ah TIL

**Comment by noisecapella** [comment]
Anything else here?

**Comment by aliceriot** [comment]
Haven't looked at it since you updated, I can take a look now.

**Comment by noisecapella** [comment]
Thanks!
